File: src/main/java/org/artembogomolova/demo/webapp/dao/repo/IUserRepository.java
package org.artembogomolova.demo.webapp.dao.repo;

import org.artembogomolova.demo.webapp.domain.auth.User;
import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface IUserRepository extends PagingAndSortingRepository<User, Long> {

}

File: app/src/main/kotlin/org/artembogomolova/demo/webapp/main/dao/repo/IUserRepository.kt
package org.artembogomolova.demo.webapp.main.dao.repo

import org.artembogomolova.demo.webapp.main.domain.auth.User
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.stereotype.Repository

@Repository
interface IUserRepository : PagingAndSortingRepository<User?, Long?> {
    fun findByLogin(login: String?): User?
}